test/unit/app/services/service_listeners/publishing_api_associated_documents_test.rb

```diff
@@ -268,15 +268,69 @@ class Publish < PublishingApiAssociatedDocumentsTest
           "en",
         )
 
+        call(new_edition)
+      end
+
+      test "with an office on the old version that remains on the new version of an editionable worldwide organisation it does not publish a redirect for the office" do
+        worldwide_organisation = create(:published_editionable_worldwide_organisation, :with_main_office)
+
+        new_edition = worldwide_organisation.create_draft(create(:writer))
+        new_edition.minor_change = true
+        new_edition.submit!
+        new_edition.publish!
+
+        old_office = worldwide_organisation.main_office
+
         PublishingApiRedirectWorker.any_instance.expects(:perform).with(
-          old_office.contact.content_id,
+          old_office.content_id,
           new_edition.search_link,
           "en",
+        ).never
+
+        call(new_edition)
+      end
+
+      test "with a contact on the old version of an editionable worldwide organisation publishes gone for the contact" do
+        worldwide_organisation = create(:published_editionable_worldwide_organisation, :with_main_office)
+
+        new_edition = worldwide_organisation.create_draft(create(:writer))
+        new_edition.main_office.destroy!
+        new_edition.minor_change = true
+        new_edition.submit!
+        new_edition.publish!
+
+        old_office = worldwide_organisation.main_office
+
+        PublishingApiGoneWorker.any_instance.expects(:perform).with(
+          old_office.contact.content_id,
+          nil,
+          nil,
+          "en",
         )
 
         call(new_edition)
       end
 
+      test "with a contact on the old version that remains on the new version of an editionable worldwide organisation it does not publish gone for the contact" do
+        worldwide_organisation = create(:published_editionable_worldwide_organisation, :with_main_office)
+
+        new_edition = worldwide_organisation.create_draft(create(:writer))
+        new_edition.minor_change = true
+        new_edition.submit!
+        new_edition.publish!
+
+        old_office = worldwide_organisation.main_office
+
+        PublishingApiGoneWorker.any_instance.expects(:perform).with(
+          old_office.contact.content_id,
+          nil,
+          nil,
+          "en",
+        ).never
```

Comment:
This `any_instance`/`never` pattern feels quite awkward to me, but I'm not familiar enough with Whitehall testing to make an alternative suggestion. I assume we're just limited to something a lot like this by the shape of the code?

Reply:
It's definitely a pattern we use elsewhere in Whitehall.  I'm not sure there's any other ideal solution really.